config_test.go

```diff
@@ -33,6 +33,205 @@ func TestEmptyClientIDConfigValidates(t *testing.T) {
 	}
 }
 
+func TestNetConfigValidates(t *testing.T) {
+	tests := []struct {
+		name string
+		cfg  func() *Config // resorting to using a function as a param because of internal composite structs
```

Comment:
change the method sig to `func(*Config)` and have the loop create the new config each time; will reduce the size of all the function bodies

Reply:
Thanks. Updated.